### PR TITLE
 [C4GT] Network-Authorization: Added test for InvalidNodeIdentifier

### LINKED ISSUE #308: [C4GT] Network-Authorization: Add tests for `InvalidNodeIdentifier`
## Description
Subtask under : https://github.com/cord-network/community/issues/7

New testcase can be tested under `cargo test -p pallet-network-authorization` after adding it in the code.

## Goals

- [ ] Add tests for `InvalidNodeIdentifier` for pallet/network-authorization

## Expected Outcome
- [ ] Test should assert for `InvalidNodeIdentifier` being returned properly in all the possible calls.
- [ ] If there are more than one function returning this error code, all of them should be validated either as one test case or as multiple test cases.
       
## Acceptance Criteria
NA

## Implementation Details
Look at other test cases, and add a test case for the same.
Ex: check method `add_well_known_node_works` which validates error code `NodeIdTooLong`

## Mockups / Wireframes
NA

---- 

## Product Name
CORD

## Organization Name
Dhiway

## Domain
Blockchain

## Tech Skills Needed
Rust

## Mentor(s)
@amarts 

## Complexity
[Low]

## Category
[Test]

## Sub Category
[Beginner friendly]

### PR DESCRIPTION
**Pull request by Jhanvi528**
Description:

Added a test node ID for invalid identifiers in mock.rs file.
Implemented tests for InvalidNodeIdentifier in 3 relevant functions to cover potential error occurrences.

fixes: #308

### CODE REVIEW

**Comment by vatsa287** [comment]
@Jhanvi528 Please go through README file for pre PR checks.
ex, 
```
rustfmt <filename>
```

File: pallets/node-authorization/src/mock.rs [hunk 1]
```diff
@@ -98,7 +98,7 @@ pub const TEST_NODE_4: &str = "12D3KooWFMNXNsQtDX5CS9Fw25WcH8JRwQALmKm1XJ5dKEZge
 pub const TEST_NODE_5: &str = "12D3KooWCzvmH8ehWaPttWVzj1ERB1FvMqVcqfDRvwLqzJDAqBih";
 pub const TEST_NODE_6: &str = "12D3KooW9tw9VUZkysjCwBpJo8ArH9TsgUWZhFqqUtii6VfTCvL6";
 pub const TEST_NODE_LEN: &str = "12D3KooWCzvmH8ehWaPttWVzj1ERB1FvMqVcqfDRvwLqzJDAqBih8123w";
-
+pub const TEST_NODE_7: &str = "13C3KooWCzVmH8ehW/PttWVzj1ERB1Fv0qVcqf1RvwLQzJDAqBih";
```
Comment (vatsa287) [inline pallets/node-authorization/src/mock.rs:101]: This looks like a valid Node Id. It is the accepted format of `UTF-8`. For the Invalid case, expects it to be in invalid format.

**Review by vatsa287 — CHANGES_REQUESTED** [review]

File: pallets/node-authorization/src/mock.rs (hunk 1 above)
Comment (Jhanvi528) [reply pallets/node-authorization/src/mock.rs:101]: actually i have used special character '/' to make it invalid

**Comment by Jhanvi528** [comment]
```
Build - cargo build --release

Clippy - cargo clippy --all --no-deps --all-targets --features=runtime-benchmarks -- -D warnings

Test - cargo test --release  --locked --features=runtime-benchmarks --no-fail-fast --verbose --color always --all --all-targets

```
@vatsa287 you are talking about these checks,right?

**Comment by vatsa287** [comment]
> ```
> Build - cargo build --release
> 
> Clippy - cargo clippy --all --no-deps --all-targets --features=runtime-benchmarks -- -D warnings
> 
> Test - cargo test --release  --locked --features=runtime-benchmarks --no-fail-fast --verbose --color always --all --all-targets
> ```
> 
> @vatsa287 you are talking about these checks,right?

@Jhanvi528 Yes on top of that please do `rustfmt` on changed files as well.
`rustfmt pallets/node-authorization/src/tests.rs`

**Comment by Jhanvi528** [comment]
> > ```
> > Build - cargo build --release
> > 
> > Clippy - cargo clippy --all --no-deps --all-targets --features=runtime-benchmarks -- -D warnings
> > 
> > Test - cargo test --release  --locked --features=runtime-benchmarks --no-fail-fast --verbose --color always --all --all-targets
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > @vatsa287 you are talking about these checks,right?
> 
> @Jhanvi528 Yes on top of that please do `rustfmt` on changed files as well. `rustfmt pallets/node-authorization/src/tests.rs`
image

i am getting these error while i am trying to commit the formatted code

**Comment by vatsa287** [comment]
Please do following
- Run rustfmt/ cargo +nightly only on the files which you changed.
- Commit with flag `--no-verify`.

**Comment by Jhanvi528** [comment]
> Please do following
> 
> * Run rustfmt/ cargo +nightly only on the files which you changed.
> * Commit with flag `--no-verify`.

yes done with that can u please check

**Review by vatsa287 — APPROVED** [review]
LGTM

**Comment by zeel991** [comment]
Hello @amarts , Please review this PR

**Review by amarts — APPROVED** [review]

**Pull request merged**